packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/FirebaseMessagingDelegate.kt

```diff
@@ -94,7 +95,14 @@ open class FirebaseMessagingDelegate(protected val context: Context) : FirebaseM
     val notification = createNotification(remoteMessage)
     DebugLogging.logNotification("FirebaseMessagingDelegate.onMessageReceived: notification", notification)
     NotificationsService.receive(context, notification)
-    // background tasks are separate from the main notification handling flow; they are executed through task-manager
+    runTaskManagerTasks(remoteMessage)
+  }
+
+  private fun runTaskManagerTasks(remoteMessage: RemoteMessage) {
+    // getTaskServiceImpl() has a side effect:
+    // the TaskService constructor calls restoreTasks which then constructs a BackgroundRemoteNotificationTaskConsumer,
+    // and the getBackgroundTasks() call below doesn't return an empty collection.
+    TaskServiceProviderHelper.getTaskServiceImpl(context.applicationContext)
```

Comment:
Should we rename `getTaskServiceImpl` to something more sensible? :) What about `getOrCreateTaskService`, `ensureTaskServiceStarted` - or something else?

Reply:
yes, I'm for it. `ensureTaskServiceStarted` sounds good to me, or `ensureTaskServiceReady`?
Does the task service have a concept of running / being stopped? If not, I'd not use the word "started".

Reply:
Valid point :) it is not `running` - it is `loaded` or not - so `ensureTaskServiceLoaded` might be the most exact naming? Could we include the change in the PR, since this is the only code using it at this point?

Reply:
most likely I'll open a separate PR; I'm not sure how quickly this'll be merged.

Reply:
may do this later